diff --git a/BE/src/main/java/com/bibbidi/wedding/common/random/RandomShuffler.java b/BE/src/main/java/com/bibbidi/wedding/common/random/RandomShuffler.java
new file mode 100644
index 0000000000000000000000000000000000000000..456b62beec752a8c5dc8560ebf6dee6870ee3fb1
--- /dev/null
+++ b/BE/src/main/java/com/bibbidi/wedding/common/random/RandomShuffler.java
@@ -0,0 +1,17 @@
+package com.bibbidi.wedding.common.random;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RandomShuffler implements Shuffler {
+
+    @Override
+    public <T> List<T> shuffle(List<T> values) {
+        List<T> shuffled = new ArrayList<>(values);
+        Collections.shuffle(shuffled);
+        return shuffled;
+    }
+}
diff --git a/BE/src/main/java/com/bibbidi/wedding/common/random/Shuffler.java b/BE/src/main/java/com/bibbidi/wedding/common/random/Shuffler.java
new file mode 100644
index 0000000000000000000000000000000000000000..c8aba5cde5055dae60bfeb0110d223452cec39b6
--- /dev/null
+++ b/BE/src/main/java/com/bibbidi/wedding/common/random/Shuffler.java
@@ -0,0 +1,8 @@
+package com.bibbidi.wedding.common.random;
+
+import java.util.List;
+
+public interface Shuffler {
+
+    <T> List<T> shuffle(List<T> values);
+}
